packages/babel-plugin-transform-for-of/src/index.js

```diff
@@ -48,8 +48,19 @@ export default declare((api, options) => {
             );
           }
 
-          const block = t.toBlock(body);
-          block.body.unshift(assignment);
+          let blockBody;
+          const body = path.get("body");
+          if (
+            body.isBlockStatement() &&
+            Object.keys(path.getBindingIdentifiers()).some(id =>
```

Comment:
Can you filter the binding identifiers `id` by
```js
id.type !== "var"
```
So we don't add extra block statement when the for binding is of `var` kind.
```js
for (var [head, ...tail] of array) {
  const head = 1;
  console.log(tail);
  console.log(head);
}
```

Reply:
Don't we still need the block? This code would be invalid:

```js
for (let _i = 0, _array = array; _i < _array.length; _i++) {
  var [head, ...tail] = _array[_i];
  const head = 1;
  console.log(tail);
  console.log(head);
}
```